test/gtest/db_sync.cpp

```diff
@@ -563,7 +564,16 @@ void CheckDynamicFDBEntry(size_t thread_index,
                     auto program_file = miopen::make_object_file_name(kern.kernel_file).string();
                     ASSERT_TRUE(!miopen::EndsWith(kern.kernel_file, ".mlir"))
                         << "MLIR detected in dynamic solvers";
-                    compile_options += " -mcpu=" + handle.GetDeviceName();
+                    if(miopen::EndsWith(kern.kernel_file, ".s"))
+                    {
+                        compile_options +=
+                            " -mcpu=" +
+                            miopen::LcOptionTargetStrings{handle.GetTargetProperties()}.targetId;
+                    }
+                    else
+                    {
+                        compile_options += " -mcpu=" + handle.GetDeviceName();
+                    }
```

Comment:
```suggestion
                    compile_options += " -mcpu=";
                    if(miopen::EndsWith(kern.kernel_file, ".s"))
                    {
                        compile_options +=
                            miopen::LcOptionTargetStrings{handle.GetTargetProperties()}.targetId;
                    }
                    else
                    {
                        compile_options += handle.GetDeviceName();
                    }
```
[R] For me it looks a bit better.